src/plugins/discover/public/application/view_alert/view_alert_route.tsx

```diff
@@ -0,0 +1,203 @@
+      try {
+        return await data.search.searchSource.create(fetchedAlert.params.searchConfiguration);
+      } catch (error) {
+        const errorTitle = i18n.translate('discover.viewAlert.searchSourceErrorTitle', {
+          defaultMessage: 'Error fetching search source',
+        });
+        displayError(errorTitle, error.message, toastNotifications);
+      }
+    };
+
+    const showDataViewFetchError = (alertId: string) => {
+      const errorTitle = i18n.translate('discover.viewAlert.dataViewErrorTitle', {
+        defaultMessage: 'Error fetching data view',
+      });
+      displayError(
+        errorTitle,
+        new Error(`Data view failure of the alert rule with id ${alertId}.`).message,
+        toastNotifications
+      );
+    };
+
+    const navigateToResults = async () => {
+      const fetchedAlert = await fetchAlert();
+      if (!fetchedAlert) {
+        history.push(DISCOVER_MAIN_ROUTE);
+        return;
+      }
+
+      const calculatedChecksum = getCurrentChecksum(fetchedAlert.params);
+      if (openConcreteAlert && calculatedChecksum !== queryParams.checksum) {
+        displayRuleChangedWarn(toastNotifications);
+      } else if (openConcreteAlert && calculatedChecksum === queryParams.checksum) {
+        displayPossibleDocsDiffInfoAlert(toastNotifications);
+      }
+
+      const fetchedSearchSource = await fetchSearchSource(fetchedAlert);
+      if (!fetchedSearchSource) {
+        history.push(DISCOVER_MAIN_ROUTE);
+        return;
+      }
+
+      const dataView = fetchedSearchSource.getField('index');
+      const timeFieldName = dataView?.timeFieldName;
+      if (!dataView || !timeFieldName) {
+        showDataViewFetchError(fetchedAlert.id);
+        history.push(DISCOVER_MAIN_ROUTE);
+        return;
+      }
+
+      let timeRange: TimeRange;
+      if (openConcreteAlert) {
+        timeRange = { from: queryParams.from, to: queryParams.to };
+      } else {
+        const filter = getTime(dataView, {
+          from: `now-${fetchedAlert.params.timeWindowSize}${fetchedAlert.params.timeWindowUnit}`,
+          to: 'now',
+        });
+        timeRange = {
+          from: filter?.query.range[timeFieldName].gte,
+          to: filter?.query.range[timeFieldName].lte,
+        };
+      }
+
+      const state: DiscoverAppLocatorParams = {
+        query: fetchedSearchSource.getField('query') || data.query.queryString.getDefaultQuery(),
+        indexPatternId: dataView.id,
+        timeRange,
+      };
+      const filters = fetchedSearchSource.getField('filter');
+      if (filters) {
+        state.filters = filters as Filter[];
+      }
+      await locator.navigate(state);
+    };
+
+    navigateToResults();
+  }, [
+    toastNotifications,
+    data.query.queryString,
+    data.search.searchSource,
+    core.http,
+    locator,
+    id,
+    queryParams,
+    history,
+    openConcreteAlert,
+  ]);
+
+  return null;
```

Comment:
Is there a better way to do this? Basically the whole point of this route is to redirect to navigate to other places.  The definition of the Route says "its most basic responsibility is to render some UI when its path matches the current URL", which we are not doing here. Should this be a `Redirect` instead?

Reply:
If you see better solution, please let me know. For now the following looks a bit weak.

```
const fullPath = await locator.getUrl(state);
const [, url] = fullPath.split('discover#');
setRedirectUrl(url);
```